src/Uno.UI/LayoutProvider.Android.cs

```diff
@@ -0,0 +1,133 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Android.App;
+using Android.Graphics;
+using Android.Graphics.Drawables;
+using Android.Util;
+using Android.Views;
+using Android.Widget;
+using Windows.Foundation;
+using Rect = Android.Graphics.Rect;
+
+namespace Uno.UI
+{
+	public class LayoutProvider
+	{
+		public delegate void LayoutChangedListener(Rect statusBar, Rect keyboard, Rect navigationBar);
+
+		public static LayoutProvider Instance { get; private set; }
+
+		public event LayoutChangedListener LayoutChanged;
+
+		public Rect StatusBarLayout { get; private set; } = new Rect(0, 0, 0, 0);
+		public Rect KeyboardLayout { get; private set; } = new Rect(0, 0, 0, 0);
+		public Rect NavigationBarLayout { get; private set; } = new Rect(0, 0, 0, 0);
+
+		private readonly Activity _activity;
+		private readonly GlobalLayoutProvider _adjustNothingLayoutProvider, _adjustResizeLayoutProvider;
+
+		internal LayoutProvider(Activity activity)
+		{
+			this._activity = activity;
+
+			_adjustNothingLayoutProvider = new GlobalLayoutProvider(activity, null)
+			{
+				SoftInputMode = SoftInput.AdjustNothing | SoftInput.StateUnchanged,
+				InputMethodMode = InputMethod.NotNeeded,
+			};
+			_adjustResizeLayoutProvider = new GlobalLayoutProvider(activity, MeasureLayout)
+			{
+				SoftInputMode = SoftInput.AdjustResize | SoftInput.StateUnchanged,
+				InputMethodMode = InputMethod.Needed,
+			};
+
+			Instance = this;
+		}
+
+		// lifecycle management
+		internal void Start(View view)
+		{
+			if (view?.WindowToken != null)
+			{
+				_adjustNothingLayoutProvider.Start(view);
+				_adjustResizeLayoutProvider.Start(view);
+			}
+		}
+		internal void Stop()
+		{
+			_adjustNothingLayoutProvider.Stop();
+			_adjustResizeLayoutProvider.Stop();
+		}
+
+		private void MeasureLayout(PopupWindow sender)
```

Comment:
You tested landscape mode right? Is the navigation bar invariably at the bottom of the screen?

Reply:
The keyboard layout detection was not support in landscape mode, and still isn't. On the few phones that I checked the soft navigation bar are always placed on the side:
- On pixel, s8: always on the side of where earplug jack/charging port is normally
- On HuaWei, lg nexeus: always on the right side of current orientation